shell: modules: promote edac mem shell subcommand to root shell command

henrikbrixandersen opened the pull request:
Promote the "edac mem" shell subcommand to a generic "devmem" root shell command. This command is useful for poking around registers and memory outside of the EDAC drivers.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen commented:
@finikorg I am not able to verify that the new "devmem" command works on MMU/PCIe hardware. Can you verify?

jakub-uC commented on subsys/shell/modules/devmem_service.c:
```diff
@@ -0,0 +1,109 @@
+#if defined(CONFIG_MMU) || defined(CONFIG_PCIE)
+	device_map((mm_reg_t *)&addr, phys_addr, 0x100, K_MEM_CACHE_NONE);
+
+	shell_fprintf(sh, SHELL_NORMAL, "Mapped 0x%lx to 0x%lx\n",
```
You can consider using shell_print(shell, "Mapped 0x%lx to 0x%lx", phys_addr, addr); to fit it in one line

henrikbrixandersen replied:
Done, thanks.

jakub-uC reviewed (approved):
Basically, I am fine with the change. Maybe it is worth mentioning in the shell documentation in the build-in commands section?

henrikbrixandersen commented:
> Basically, I am fine with the change. Maybe it is worth mentioning in the shell documentation in the build-in commands section?

That section only seems to cover "real" built-in shell commands (e.g. `clear`, `history`, etc). This is not really a shell built-in command.

henrikbrixandersen committed:
shell: modules: promote edac mem shell subcommand to root shell command

Promote the "edac mem" shell subcommand to a generic "devmem" root shell
command. This command is useful for poking around registers and memory
outside of the EDAC drivers.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

nordic-krch commented on subsys/shell/modules/devmem_service.c:
```diff
@@ -0,0 +1,108 @@
+	return memory_write(sh, addr, width, value);
+}
+
+SHELL_CMD_REGISTER(devmem, NULL, "Read/write physical memory\""
```
Can you extend help a bit, at least add info that address and value is in hex and width in bits.

finikorg reviewed (approved):
Thanks. I was thinking to move it same way. Probably more documentation might help, I mentioned the format should be similar to busybox's devmem.
